python/ambassador/resource.py

```diff
@@ -181,10 +181,11 @@ def from_dict(cls: Type[R], rkey: str, location: str, serialization: Optional[st
         # the correct type.
         ambassador = sys.modules['ambassador']
 
-        resource_class: Type[R] = getattr(ambassador, attrs['kind'], None)
+        resource_class: Optional[Type[R]] = getattr(ambassador, attrs['kind'], None)
 
         if not resource_class:
             resource_class = getattr(ambassador, 'AC' + attrs[ 'kind' ], cls)
+        assert resource_class
```

Comment:
@kflynn I'm not sure why mypy isn't catching this on your box (it does catch it on my box), and why it wasn't catching it before. But mypy was right; the type annotation was wrong. It needed an `Optional[…]` wrapper, and then that `Optional` trickles down to need us to tell it that `resource_class` is truthy after the `if`.